### ISSUE TITLE
Main file defined in project settings (TEXroot) depends on cwd if relative paths are used

### ISSUE THREAD

**Issue by neelabhg** [SublimeText/LaTeXTools#630]
#### Problem
In the case when:

1. The main file is defined in project settings using a relative path, and
2. The source file, from which a command is invoked, is in a directory different than that of the main TeX root file (for example, `chapters/ch1.tex`),

the following issues may be experienced:

* If the `TEXroot` is not relative to the source file (for example `TEXroot: "main.tex"`), then the main file is not found, and the source file is taken to be the root file, causing errors during compilation. Viewing the PDF file also does not work, because it tries to launch the wrong PDF (for example, `chapters/ch1.pdf`, which most likely does not exist).

* If the `TEXroot` is relative to the source file (for example, `TEXroot: "../main.tex"`), then the main file is found and compilation succeeds, but when launching the PDF viewer after compilation, the main file is not found (because the cwd at that point is that of the root file itself), and so the wrong pdf is launched (for example, `chapters/ch1.pdf`). Viewing the PDF file directly, however, does work because the main file is set correctly.

#### Workaround
Currently, the solution is to use an absolute path in the project settings, but it's not ideal because the `.sublime-project` file is usually checked in to version control.

#### Possible solution
The ideal solution would be to resolve paths relative to the project directory, similar to the behavior of the `folders` section of [Sublime Text project definition files](https://www.sublimetext.com/docs/2/projects.html).

One way to do this could be to use the project directory as the base directory in [`getTeXRoot.py#L26`](https://github.com/SublimeText/LaTeXTools/blob/6bb8ab9cfcb3fc4dcf1a47c8d6c1ea0db428f95b/getTeXRoot.py#L26). So, instead of
```py
root = os.path.abspath(view.settings().get('TEXroot'))
```

we can have something like
```py
root = os.path.abspath(os.path.join(get_project_root(view), view.settings().get('TEXroot')))
```

Getting the project root seems tricky. This is because while [Sublime Text 3's API supports this](https://www.sublimetext.com/docs/3/api_reference.html#sublime.Window), there doesn't seem to be a way to access the same in Sublime Text 2. A workaround could be to get the list of project folders and pick the first one as the project root, as suggested in this [forum discussion](https://www.sublimetext.com/forum/viewtopic.php?f=6&t=1253). A downside to this workaround could be that the project folder may not necessarily be the first folder in the list of folders, if there are multiple folders open inside the window. With that in mind, a possible implementation of `get_project_root` could be:

```py
def get_project_root(view):
    if (_ST3):
        return os.path.dirname(view.window().project_file_name())
    return view.window().folders()[0]
```

I hope this issue makes sense, and the provided solution is helpful. Let me know if I can help out in any other way, or if there are any questions I can answer.

**Comment by ig0774** [comment]
What you've highlighted has been the reason that we don't (technically) support relative paths in the `TEXroot` setting (see also #152, #485). OTOH, I just submitted a PR (#629) that addresses this, which you may want to take a look at. This will eventually be merged once I'm satisfied it doesn't break anything.

**Comment by ig0774** [comment]
Once again, thank you for your very thorough and well thought-out report. The fix in #629 was committed to the repository and released today. Could you, when you have a chance, update your version of LaTeXTools and verify that the behaviour works? Thanks!
